sm/tedge_agent/src/restart_operation_handler.rs

```diff
@@ -0,0 +1,112 @@
+pub mod restart_operation {
+
+    use crate::error::AgentError;
+    use std::{fs::File, fs::OpenOptions, io::Read, io::Write, path::Path};
+    use time::OffsetDateTime;
+    use tracing::error;
+
+    const SLASH_RUN_PATH_TEDGE_AGENT_RESTART: &str = "/run/tedge_agent_restart";
```

Comment:
I think better to create a directory owned by `tegde-agent` during the installation script, and create a file under the directory. Then, the `tedge_agent` run by `tedge-agent` user can create a run file. For example, `/run/tedge_agent/restart`.

Reply:
But on almost all systems `/run` is cleared on system restart @rina23q